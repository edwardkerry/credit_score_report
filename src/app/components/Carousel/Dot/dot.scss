.carousel__dot {
  display: block;
  width: 30px;
  height: 30px;
  border-radius: 50%;
  background-color: #c9c9d8;
  cursor: pointer;
  opacity: .15;
  &:hover {
    opacity: .5;
  }
  &--active {
    &,
    &:hover {
      opacity: .75;
    }
  }
}