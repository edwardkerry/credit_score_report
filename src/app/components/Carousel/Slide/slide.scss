.carousel__slide {
  position: relative;
  display: none;
  list-style-type: none;
  text-align: center;
  right: -100px;
  &--active {
    animation: slide 0.5s forwards;
    display: block;
  }
  @keyframes slide {
    100% {
      right: 0
    }
  }
}
