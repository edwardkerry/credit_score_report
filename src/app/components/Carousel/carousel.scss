.carousel__container {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  width: 100%;
  ul {
    padding: 0;
    margin: 0;
    list-style-type: none;
  }
}

.carousel__dots {
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin-top: 20px;
  li {
    &:nth-of-type(n + 2) {
      margin-left: 9px;
    }
  }
}