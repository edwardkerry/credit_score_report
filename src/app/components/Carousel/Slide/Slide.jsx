import React, { Component } from 'react';
import { object, bool } from 'prop-types';
import bemHelper from '../../../utils/bem';
import './slide.scss';

const cn = bemHelper({ block: 'carousel' });

const Slide = ({content, active} ) => (
  <li
    className={
      active ? cn('slide', 'active') : cn('slide', null)
    }
  >
  {content}
  </li>
)

Slide.propTypes = {
  content: object.isRequired,
  active: bool
}

Slide.defaultProps = {
  active: false
}

export default Slide;