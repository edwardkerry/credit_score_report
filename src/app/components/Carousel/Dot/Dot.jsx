import React, { Component } from 'react';
import { func, bool } from 'prop-types';
import bemHelper from '../../../utils/bem';
import '../carousel.scss';

const cn = bemHelper({ block: 'carousel' });

const Dot = ({active, select}) => (
  <li>
    {active ? 
      <a className={ cn('dot', 'active')}/> :
      <a className={cn('dot', 'null')}
        onClick={select}
      /> 
    } 
  </li>
)

Dot.propTypes = {
  active: bool,
  select: func.isRequired
}

Dot.defaultProps = {
  active: false
}

export default Dot;