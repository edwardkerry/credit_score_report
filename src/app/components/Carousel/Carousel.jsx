import React, { Component } from 'react';
import { array } from 'prop-types';
import bemHelper from '../../utils/bem';
import Slide from './Slide/Slide.jsx';
import Dot from './Dot/Dot.jsx';
import './carousel.scss';
import { removeDotSegments } from 'uri-js';

const cn = bemHelper({ block: 'carousel' });

class Carousel extends Component {

  constructor(props) {
    super(props);
    this.state = {
      activeSlideIndex: 0
    };

    this.keyCount = 0;
    this.getKey = this.getKey.bind(this);
    this.selectSlide = this.selectSlide.bind(this)
  };
  
  selectSlide(index) {
    this.setState({ activeSlideIndex: index });
  };

  getKey() {
    return `CarouselKey${this.keyCount ++}`
  }

  render() {
    return (
      <div className={cn('container', null)}>
        <ul className={cn('slides', null)}>
          {this.props.slides.map((slide, index) =>{
           return(<Slide 
              content={slide}
              active={index == this.state.activeSlideIndex}
              key={this.getKey()}
            />)
          })}
        </ul>

        <ul className={cn('dots', null)}>
          {this.props.slides.map((slide, index) => {
            return(<Dot
              select={() => this.selectSlide(index)}
              active={index == this.state.activeSlideIndex}
              key={this.getKey()}
            />) 
          })}
        </ul>
      </div>
    );
  };
};

Carousel.propTypes = {
  slides: array
};

export default Carousel;