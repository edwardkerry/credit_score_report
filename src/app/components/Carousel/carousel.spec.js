import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Carousel from './Carousel.jsx';
import Slide from './Slide/Slide.jsx';
import Dot from './Dot/Dot.jsx';

describe('Carousel', () => {
  const componentOne = (<h1>Example 1</h1>)
  const componentTwo = (<h2>Example 2</h2>)
  const componentThree = (<h3>Example 3</h3>)
  const passedComponents = [componentOne, componentTwo, componentThree]
  it('renders', () => {
    const component = renderer.create(<Carousel slides={passedComponents} />)
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders the passed components as slides', () => {
    const component = shallow(<Carousel slides={passedComponents} />)
    expect(component.find('.carousel__slides').children().length).toEqual(passedComponents.length)
  })

  it('renders a selection dot for each passed component', () => {
    const component = shallow(<Carousel slides={passedComponents} />)
    expect(component.find('.carousel__dots').children().length).toEqual(passedComponents.length)
  })

  describe('#getKey', () => {
    it('increments the keyCount to generate a unique ID', () => {
      const component = shallow(<Carousel slides={[]}/>).instance()
      expect(component.getKey()).toEqual('CarouselKey0')
      expect(component.getKey()).toEqual('CarouselKey1')
    });
  });

  describe('#selectSlide', () => {
    it('sets the passed value as the current active slide index')
    const currentIndex = 999
    const component = mount(<Carousel slides={[]} />).instance()
    component.selectSlide(currentIndex)
    expect(component.state.activeSlideIndex).toEqual(999)
  })
});