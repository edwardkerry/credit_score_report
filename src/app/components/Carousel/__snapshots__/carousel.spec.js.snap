// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel renders 1`] = `
<div
  className="carousel__container"
>
  <ul
    className="carousel__slides"
  >
    <li
      className="carousel__slide carousel__slide--active"
    >
      <h1>
        Example 1
      </h1>
    </li>
    <li
      className="carousel__slide"
    >
      <h2>
        Example 2
      </h2>
    </li>
    <li
      className="carousel__slide"
    >
      <h3>
        Example 3
      </h3>
    </li>
  </ul>
  <ul
    className="carousel__dots"
  >
    <li>
      <a
        className="carousel__dot carousel__dot--active"
      />
    </li>
    <li>
      <a
        className="carousel__dot carousel__dot--null"
        onClick={[Function]}
      />
    </li>
    <li>
      <a
        className="carousel__dot carousel__dot--null"
        onClick={[Function]}
      />
    </li>
  </ul>
</div>
`;
